src/legacy/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -710,20 +704,29 @@ function discoverController(
 
     $scope.updateTime();
 
+    $scope.searchSource.cancelQueued();
```

Comment:
might be worth adding a short comment here for future reference, it case it isn't immediately obvious why we are doing this